src/cli-scan.ts

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+
+import fastGlob from 'fast-glob';
+import yargs from 'yargs-parser';
+import { logger } from './logger';
+import colors from 'colors';
+import { ADMIN_DASH } from './constants';
+import { scanPackageJson } from './plugins/scanPackageJson';
+import { GraphQLClient } from 'graphql-request';
+import { ADD_SILO_DISCOVERY_RESULTS } from './gqls';
+
+const SUPPORTED_FILE_SCANS = ['package.json'];
+const IGNORE_DIRS = ['node_modules', 'serverless-build'];
+
+/**
+ * Sync data silo configuration from Transcend down locally to disk
+ *
+ * Dev Usage:
+ * yarn ts-node ./src/cli-scan.ts --scanPath=./myJavascriptProject --auth=asd123 --ignoreDirs=node_modules,serverless-build
+ *
+ * Standard usage
+ * yarn tr-scan  --scanPath=./myJavascriptProject --auth=asd123
+ */
+async function main(): Promise<void> {
+  // Parse command line arguments
+  const {
+    scanPath = '.',
+    ignoreDirs = '',
+    transcendUrl = 'https://api.transcend.io',
+    pluginId = '',
+    auth,
+  } = yargs(process.argv.slice(2));
+
+  // // Ensure auth is passed
+  if (!auth) {
+    logger.error(
+      colors.red(
+        'A Transcend API key must be provided. You can specify using --auth=asd123',
+      ),
+    );
+    process.exit(1);
+  }
+
+  // Create a GraphQL client
+  // eslint-disable-next-line global-require
+  const { version } = require('../package.json');
+  const client = new GraphQLClient(`${transcendUrl}/graphql`, {
+    headers: {
+      Authorization: `Bearer ${auth}`,
+      version,
+    },
+  });
+  const dirsToIgnore = [...IGNORE_DIRS, ...ignoreDirs.split(',')];
+  const filesToScan = await fastGlob(
```

Comment:
to make this more extensible, i was thinking that we'd exchange the pluginId for the dataSilo.type and then create a map in this codebase from dataSilo.type -> function that fetches silo suggestions, then we just pop on new files like `src/plugins/scanPackageJson.ts`

Reply:
@michaelfarrell76 to make sure I understand you:
1. We'd create a mapping of integrationType to parsing/plugin functions (e.g. `{ javascriptPackageJson: scanPackageJson }`) 
2. The user would pass in the integrationType instead of the pluginId
3. We'd use the integrationType to resolve the pluginId by querying the backend GraphQL
4. We'd use the integrationType to resolve the scanning function to invoke

Reply:
ok one complication i see is that the function that fetches silos suggestions can have wildly different input types so the typing feels hard 😭 

if we're just scanning static files then i think it's fine to expect a file path, but maybe in the future we want to build out other push-based plugins that execute a DB dump or query against a local DB and upload that to transcend, then the input to that function would be pretty different?

Reply:
since there's no clean easy way to query for an active pluginID based on integrationType yet i will revisit this later

Reply:
sorry minh did see this til now.

I would recommend:

1. We'd create a mapping of integrationType to parsing/plugin functions (e.g. { javascriptPackageJson: scanPackageJson })
~2. The user would pass in the integrationType instead of the pluginId~ -> continue to pass in pluginId but call a GraphQL query to find the integrationType for that pluginId. (we may need to update one of our queries)
~We'd use the integrationType to resolve the pluginId by querying the backend GraphQL~
~We'd use the integrationType to resolve the scanning function to invoke~

Reply:
im also not sure we should expect an exact file path. that definitely makes it harder for monorepos. 

my hope was that you could just pass in the root folder or a specific package.json file and then the mapping function does a glob on that folder to find all package.json files underneath it, or take the file if it is a package,json.

Reply:
> ok one complication i see is that the function that fetches silos suggestions can have wildly different input types so the typing feels hard 😭

wouldnt worry too much here, would just have two things a) the raw data and b) the field that should be used to predict saas tool

Reply:
@michaelfarrell76 hmm but how to obtain the raw data would vary a lot between different saas tools, no? Can you take a look at what I have now?

`scanPackageJson` does some extra work (like filtering out directories to ignore, and parsing to JSON) that would be hard to capture if all we're doing is passing in the raw data i think

Reply:
Also the file to glob for would vary based on the plugin/file we're scanning

Reply:
i was thinking the interface is like: (fileOrFolderPath: string) => Suggestion[]